promunifi/usg.go

```diff
@@ -0,0 +1,186 @@
+package promunifi
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"golift.io/unifi"
+)
+
+type usg struct {
+	Uptime         *prometheus.Desc
+	Temperature    *prometheus.Desc
+	TotalMaxPower  *prometheus.Desc
+	FanLevel       *prometheus.Desc
+	TotalTxBytes   *prometheus.Desc
+	TotalRxBytes   *prometheus.Desc
+	TotalBytes     *prometheus.Desc
+	NumSta         *prometheus.Desc
+	UserNumSta     *prometheus.Desc
+	GuestNumSta    *prometheus.Desc
+	NumDesktop     *prometheus.Desc
+	NumMobile      *prometheus.Desc
+	NumHandheld    *prometheus.Desc
+	Loadavg1       *prometheus.Desc
+	Loadavg5       *prometheus.Desc
+	Loadavg15      *prometheus.Desc
+	MemBuffer      *prometheus.Desc
+	MemTotal       *prometheus.Desc
+	MemUsed        *prometheus.Desc
+	CPU            *prometheus.Desc
+	Mem            *prometheus.Desc
+	WanRxPackets   *prometheus.Desc
+	WanRxBytes     *prometheus.Desc
+	WanRxDropped   *prometheus.Desc
+	WanRxErrors    *prometheus.Desc
+	WanTxPackets   *prometheus.Desc
+	WanTxBytes     *prometheus.Desc
+	LanRxPackets   *prometheus.Desc
+	LanRxBytes     *prometheus.Desc
+	LanRxDropped   *prometheus.Desc
+	LanTxPackets   *prometheus.Desc
+	LanTxBytes     *prometheus.Desc
+	WanRxBroadcast *prometheus.Desc
+	WanRxBytesR    *prometheus.Desc
+	WanRxMulticast *prometheus.Desc
+	WanSpeed       *prometheus.Desc
+	WanTxBroadcast *prometheus.Desc
+	WanTxBytesR    *prometheus.Desc
+	WanTxDropped   *prometheus.Desc
+	WanTxErrors    *prometheus.Desc
+	WanTxMulticast *prometheus.Desc
+	WanBytesR      *prometheus.Desc
+	Latency        *prometheus.Desc
+	Runtime        *prometheus.Desc
+	XputDownload   *prometheus.Desc
+	XputUpload     *prometheus.Desc
+}
+
+func descUSG(ns string) *usg {
+	if ns += "_usg_"; ns == "_usg_" {
+		ns = "usg_"
+	}
+	labels := []string{"site_name", "mac", "model", "name", "serial", "site_id",
+		"type", "version", "ip"}
+	labelWan := append([]string{"port"}, labels...)
+
+	return &usg{
+		Uptime:         prometheus.NewDesc(ns+"uptime", "Uptime", labels, nil),
+		TotalTxBytes:   prometheus.NewDesc(ns+"transmit_bytes_total", "Total Transmitted Bytes", labels, nil),
+		TotalRxBytes:   prometheus.NewDesc(ns+"receive_bytes_total", "Total Received Bytes", labels, nil),
+		TotalBytes:     prometheus.NewDesc(ns+"transferred_bytes_total", "Total Bytes Transferred", labels, nil),
+		NumSta:         prometheus.NewDesc(ns+"num_stations_total", "Number of Stations", labels, nil),
```

Comment:
This seems like a gauge, so it shouldn't end in `_total`. Also `num` is redundant.

```suggestion
		NumSta:         prometheus.NewDesc(ns+"stations", "Number of Stations", labels, nil),
```

Reply:
there's a lot wrong like this..

Reply:
I realize num is redundant, but it lines up the "user/device" counters very nice to keep that prefix on all of these.

Reply:
I would suggest removing `num` from all of them. :smile: